rust/pst/contract/implementation/src/actions/transfer.rs

```diff
@@ -1,36 +1,40 @@
-use warp_pst::action::{ActionResult, HandlerResult, Transfer};
-use warp_pst::error::ContractError;
-use warp_pst::state::{State};
-use warp_wasm_utils::contract_utils::js_imports::{log, SmartWeave, Transaction};
+use warp_pst::{
+    action::{ActionResult, Transfer},
+    error::ContractError,
+    state::State,
+};
+use warp_wasm_utils::contract_utils::{
+    handler_result::HandlerResult::NewState,
+    js_imports::{log, SmartWeave, Transaction},
+};
 
-use super::{Actionable};
+use super::Actionable;
 
 impl Actionable for Transfer {
-     fn action(self, caller: String, mut state: State) -> ActionResult {
+    fn action(self, _caller: String, mut state: State) -> ActionResult {
```

Comment:
general comment - shouldn't the SmartWeave global object be passed here as an argument? this would greatly simplify the mocking - in case one would like to write tests for the contract's functions directly in Rust

Reply:
At this point, this is up to the contract author. What we standardize is the `handle` method interface.